rust/runfiles/runfiles.rs

```diff
@@ -261,10 +318,31 @@ fn parse_repo_mapping(path: PathBuf) -> Result<RepoMapping> {
         if parts.len() < 3 {
             return Err(RunfilesError::RepoMappingInvalidFormat);
         }
-        repo_mapping.insert((parts[0].into(), parts[1].into()), parts[2].into());
+
+        let source_repo = parts[0];
+        let apparent_name = parts[1];
+        let target_repo = parts[2];
+
+        // Check if this is a prefix entry (ends with '*')
+        // The '*' character is terminal and marks a prefix match entry
+        if let Some(prefix) = source_repo.strip_suffix('*') {
+            prefixes.push((
+                (prefix.to_owned(), apparent_name.to_owned()),
+                target_repo.to_owned(),
+            ));
+        } else {
+            exact.insert(
+                (source_repo.to_owned(), apparent_name.to_owned()),
+                target_repo.to_owned(),
+            );
+        }
     }
 
-    Ok(repo_mapping)
+    Ok(if !prefixes.is_empty() {
+        RepoMapping::Compact { exact, prefixes }
+    } else {
+        RepoMapping::Verbose(exact)
```

Comment:
Is this case needed? The compact case is meant to subsume it with an empty prefix list.